File: FinanceApp.Shared/Entities/UserTables/ForecastParameters.cs
namespace FinanceApp.Shared.Entities.UserTables
{
    public class ForecastParameters : UserTable
    {
        public double PercentageCdiLoan { get; set; }
        public double PercentageCdiFixedInteresIncometSavings { get; set; }
        public double PercentageCdiVariableIncome { get; set; }
        // Quantos % da entrada de valores no patrimonio deve ser liquido?
        public double SavingsLiquidPercentage { get; set; }
        public int MonthsSavingWarning { get; set; }

        public override void CheckInput()
        {
        }
    }
}

File: FlutterApp/finance_app/lib/models/forecast_parameters/forecast_parameters.dart
class ForecastParameters {
  double percentageCdiLoan;
  double percentageCdiFixedInteresIncometSavings;
  double percentageCdiVariableIncome;
  double savingsLiquidPercentage;
  int monthsSavingWarning;

  ForecastParameters({
    required this.percentageCdiLoan,
    required this.percentageCdiFixedInteresIncometSavings,
    required this.percentageCdiVariableIncome,
    required this.savingsLiquidPercentage,
    required this.monthsSavingWarning,
  });

  ForecastParameters.fromJson(Map<String, dynamic> json)
      : percentageCdiLoan = json['percentageCdiLoan'],
        percentageCdiVariableIncome = json['percentageCdiVariableIncome'],
        savingsLiquidPercentage = json['savingsLiquidPercentage'],
        percentageCdiFixedInteresIncometSavings =
            json['percentageCdiFixedInteresIncometSavings'],
        monthsSavingWarning = json['monthsSavingWarning'];
}
